src/main/scala/com/softwaremill/realworld/articles/ArticlesService.scala

```diff
@@ -16,7 +16,7 @@ class ArticlesService(articlesRepository: ArticlesRepository):
     .find(slug)
     .flatMap {
       case Some(a) => ZIO.succeed(a)
-      case None    => ZIO.fail(Exceptions.NotFound(s"Article with slug $slug doesn't exist."))
+      case None    => ZIO.fail(Exceptions.NotFound(s"Article with provided slug doesn't exist."))
```

Comment:
I checked that $slug (as a String) comes from the path parameter, so I think it can be included in the exception - I'll reverse that change.
I generally try to avoid situations where request body fields are directly returned in the response without sanitization due to good security practices regarding vector of attacks on the parser etc., but that's not the case here.